### PR TITLE
📝 Adding mkdocs to workflows repository

### PR DESCRIPTION
**Pull request by gaspardmoindrot**

### CODE REVIEW

File: .github/workflows/build-docs.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+      - uses: actions/cache@v2
+        with:
+          key: ${{ github.ref }}
+          path: .cache
```
[the remark this answers is not in the document]
Comment (gaspardmoindrot) [reply .github/workflows/build-docs.yaml:20]: "This code block uses the GitHub Actions Cache v2 to store and retrieve files between workflow runs, which can save time and reduce network usage." -> We don't really need it in this project, I'll take it out

File: .github/workflows/build-docs.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      - run: pip install mkdocs-material
+      - run: mkdocs build --strict
```
[the remark this answers is not in the document]
Comment (gaspardmoindrot) [reply .github/workflows/build-docs.yaml:22]: Done

File: .github/workflows/publish-docs.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      - run: pip install mkdocs-material
+      - run: mkdocs build --strict
+      - name: Setup Pages
+        uses: actions/configure-pages@v1
+      - uses: actions/upload-pages-artifact@v1
+        with:
+          path: site/
+      - name: Deploy to GitHub Pages
+        id: deployment
+        uses: actions/deploy-pages@v1
```
[the remark this answers is not in the document]
Comment (gaspardmoindrot) [reply .github/workflows/publish-docs.yaml:40]: Done

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
+# workflows
```
[the remark this answers is not in the document]
Comment (gaspardmoindrot) [reply docs/index.md:1]: Done!

File: .github/workflows/build-docs.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tcarmet) [reply .github/workflows/build-docs.yaml:20]: We may need it, to cache the pip install per example, but at the current state it's missing some config.

You can add the cache on the setup-python action above:
```yaml
      - uses: actions/setup-python@v4
         with:
           python-version: 3.x
           cache: pip
```
And remove this particular step

File: .github/workflows/build-docs.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tcarmet) [reply .github/workflows/build-docs.yaml:22]: I don't see it, are you missing a push?

File: .github/workflows/publish-docs.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tcarmet) [reply .github/workflows/publish-docs.yaml:40]: Don't see it as well.

File: mkdocs.yml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      emoji_generator: !!python/name:materialx.emoji.to_svg
+  - pymdownx.inlinehilite
+  - pymdownx.snippets
+  - pymdownx.superfences
+  - tables
+  - attr_list
+  - md_in_html
+  - meta
+  - admonition
+  - pymdownx.details
```
Comment (tcarmet) [inline mkdocs.yml:57]: New line

Comment (gaspardmoindrot) [reply mkdocs.yml]: Done

File: mkdocs.yml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      toggle:
+        icon: material/brightness-4
+        name: Switch to light mode
+  font:
+    text: Roboto
+    code: Roboto Mono
+
+nav:
+- Home: index.md
+
```
Comment (tcarmet) [inline mkdocs.yml:42]: Index the missing file here, otherwise your doc will only contain `index.md`

File: mkdocs.yml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+      toggle:
+        icon: material/brightness-4
+        name: Switch to light mode
+  font:
+    text: Roboto
+    code: Roboto Mono
+
+nav:
+- Docker build: docker-build.md
+- Known issues: known-issues.md
```
Comment (tcarmet) [inline mkdocs.yml:42]: Would be good to keep a quick intro to the site, nothing crazy, so add a `docs/index.md` file with someting like:

```markdown
# Workflows

This repository is used to share GitHub Actions reusable workflows across the organization.

```
You can add link to the original doc of what is a reusable workflow, and that's it. 

```suggestion
nav:
- Home: index.md
- Docker build: docker-build.md
- Known issues: known-issues.md
```

**Review by tcarmet — APPROVED** [review]

**Pull request merged**